### PR TITLE
feat(scope): implement Fiber roots

### PR DESCRIPTION
**Pull request by 0x706b**
#1035 

## Changelog

- feat(scope): implement Fiber roots
- feat(effect): implement `repeatUntilEffect` and `repeatUntil`
- feat(hashset): implement `from`
- test(fiber): add `roots` test

### CODE REVIEW

**Review by mikearnaldi — APPROVED** [review]

File: packages/system-next/src/Fiber/context.ts [hunk 1]
```diff
@@ -124,6 +124,8 @@ export type FiberRefLocals = Map<FiberRef.Runtime<any>, any>
 
 export const currentFiber = new AtomicReference<FiberContext<any, any> | null>(null)
 
+export const _roots = new Set<FiberContext<any, any>>()
```
Comment (IMax153) [inline packages/system-next/src/Fiber/context.ts:127]: Should we make this a `LazyValue`? @mikearnaldi @0x706b

**Review by IMax153 — CHANGES_REQUESTED** [review]

File: packages/system-next/src/Fiber/context.ts (hunk 1 above)
Comment (0x706b) [reply packages/system-next/src/Fiber/context.ts:127]: Couldn't hurt 👍🏻

Comment (mikearnaldi) [reply packages/system-next/src/Fiber/context.ts]: doesn't hurt but also it is a Set so it shouldn't be a big deal (doesn't depend on anything)

**Review by IMax153 — COMMENTED** [review]
Thank you for your work on this @0x706b !

**Pull request merged**